docs/dns.md

```diff
@@ -0,0 +1,135 @@
+# DNS Architecture in ARO HCP
+
+## Overview
+
+ARO HCP uses a dual DNS hierarchy approach with two distinct domain structures:
+
+- **Service (SVC) Zone**: Internal Azure infrastructure hosting the Resource Provider and related services
+- **Customer Experience (CX) Zone**: Customer-facing domains for Hosted Control Planes (HCPs)
+
+ARO HCP operates as a regional service with completely independent instances per Azure region. Each regional deployment creates:
+
+- A regional DNS zone under the SVC parent zone (e.g. `uksouth.{svcParentZoneName}`)
+- A regional DNS zone under the CX parent zone (e.g. `uksouth.{cxParentZoneName}`)
+
+These hierarchies have different management models:
+
+- **SVC records** are managed at infrastructure buildout time during regional deployment
+- **CX record / subzones** are dynamically managed at HCP provisioning time
+
+## DNS Zone Structure
+
+### Parent Zone Configuration
+
+The DNS hierarchy is built on two parent zones defined in configuration:
+
+- `dns.svcParentZoneName`: Parent zone for service infrastructure
+- `dns.cxParentZoneName`: Parent zone for customer-facing domains
+
+These parent zones are owned and managed by the ARO HCP team and are located in the global subscription (`global.subscription`) and global resource group (`global.rg`).
+
+### Regional Zones
+
+Regional zones are located in the regional subscription (`service.subscription`) and regional resource group (`regionRG`).
+
+Regional DNS zones follow a consistent naming pattern:
+
+```text
+{dns.regionalSubdomain}.{dns.svcParentZoneName}
+{dns.regionalSubdomain}.{dns.cxParentZoneName}
+```
+
+Examples:
+
+- `uksouth.aro-hcp.azure.com`
+- `uksouth.aroapp-hcp.io`
+
+### Environment Differentiation
+
+The staging environment shares the same parent zone as production and uses a modified naming convention to distinguish from production. This is achieved by adding the `staging` suffix to the regional subdomain configuration.
+
+```yaml
+config.yaml:
+...
+  dns:
+    regionalSubdomain: "{{ .ctx.region }}staging"
+```
+
+Example: `uksouthstaging.aroapp-hcp.io`
+
+### Zone Management and Deployment
+
+DNS zones are created as part of the infrastructure deployment process:
+
+- **Parent zones**: Managed in `global-infra.bicep`, delegation from the parent-parent zones is managed through EV2
```

Comment:
> delegation from the parent-parent zones is managed through EV2

what's the process involved here? could you add a few words clarifying this?

Reply:
added some clarification